lens-data/lens/private/syntax/stx.rkt

```diff
@@ -229,7 +247,15 @@
                   (list 1* 2* 3*))
     (check-exn #rx"expected a stx-list, given #<syntax.* 5>"
                (λ () (lens-view stx->list-lens #'5)))
+    (test-case "stx->list-lens preserves the distinction between #'(a . (b c)) and #'(a b c)"
+      (let ([one-dot-two-three (lens-set stx->list-lens #'(a . (b c)) '(1 2 3))])
+        ;; Check that the result has the shape #'(1 . (2 3)) and not #'(1 2 3)
+        (check-true (syntax? one-dot-two-three))
```

Comment:
Using `check-pred` for these tests removes some parens and gives better test failure messages